src/test/java/teammates/test/cases/webapi/SendErrorReportActionTest.java

```diff
@@ -25,35 +31,42 @@ protected String getRequestMethod() {
-        String content = "Email content";
-
         gaeSimulation.logoutUser();
 
-        ______TS("Not enough parameters");
+        ______TS("Normal case: valid report with all fields populated");
+        ErrorReportRequest report = new ErrorReportRequest(REQUEST_ID, SUBJECT, CONTENT);
+        SendErrorReportAction action = getAction(report, PARAMS);
+        JsonResult jsonResult = getJsonResult(action);
 
-        verifyHttpParameterFailure();
-        verifyHttpParameterFailure(Const.ParamsNames.ERROR_FEEDBACK_EMAIL_SUBJECT, subject);
-        verifyHttpParameterFailure(Const.ParamsNames.ERROR_FEEDBACK_REQUEST_ID, requestId);
+        String expectedLogMessage = "====== USER FEEDBACK ABOUT ERROR ======" + System.lineSeparator()
+                + "USER: Non-logged in user" + System.lineSeparator()
+                + "REQUEST ID: " + REQUEST_ID + System.lineSeparator()
+                + "SUBJECT: " + SUBJECT + System.lineSeparator()
+                + "CONTENT: " + CONTENT;
 
-        ______TS("Normal case");
+        assertEquals(HttpStatus.SC_OK, jsonResult.getStatusCode());
+        assertEquals(expectedLogMessage, action.getUserErrorReportLogMessage(report));
 
-        String[] params = {
-                Const.ParamsNames.ERROR_FEEDBACK_EMAIL_SUBJECT, subject,
-                Const.ParamsNames.ERROR_FEEDBACK_REQUEST_ID, requestId,
-        };
-        SendErrorReportAction a = getAction(content, params);
-        JsonResult r = getJsonResult(a);
+        ______TS("Failure: Invalid report with null requestId");
+        assertThrows(InvalidHttpRequestBodyException.class, () -> {
+            ErrorReportRequest badReport = new ErrorReportRequest(null, SUBJECT, CONTENT);
+            SendErrorReportAction a = getAction(badReport, PARAMS);
+            getJsonResult(a);
+        });
 
-        String expectedLogMessage = "====== USER FEEDBACK ABOUT ERROR ======" + System.lineSeparator()
-                + "USER: Non-logged in user" + System.lineSeparator()
-                + "REQUEST ID: " + requestId + System.lineSeparator()
-                + "SUBJECT: " + subject + System.lineSeparator()
-                + "CONTENT: " + content;
+        ______TS("Failure: Invalid report with null SUBJECT");
+        assertThrows(InvalidHttpRequestBodyException.class, () -> {
+            ErrorReportRequest badReport = new ErrorReportRequest(REQUEST_ID, null, CONTENT);
+            SendErrorReportAction a = getAction(badReport, PARAMS);
+            getJsonResult(a);
+        });
 
-        assertEquals(HttpStatus.SC_OK, r.getStatusCode());
-        assertEquals(expectedLogMessage, a.getUserErrorReportLogMessage());
+        ______TS("Failure: Invalid report with null CONTENT");
+        assertThrows(InvalidHttpRequestBodyException.class, () -> {
+            ErrorReportRequest badReport = new ErrorReportRequest(REQUEST_ID, SUBJECT, null);
+            SendErrorReportAction a = getAction(badReport, PARAMS);
+            getJsonResult(a);
```

Comment:
Pretty sure even `getAction` will already trigger the Exception, so this is not needed. Similarly for all other cases above.

Reply:
`getAction` alone will not trigger the exception because validation is only run when the action is executed. I changed this to call `getAction(badReport, PARAMS).execute()` instead so that the `getAndValidateRequestBody()` is run.